### DISCUSSION TITLE
Different performance on Intel and AMD CPUs

### DISCUSSION THREAD

**Discussion by peterropac** [NanoComp/meep#2727] [Q&A] [answered]
I have noticed that the simulation usually takes longer on systems with AMD CPUs compared to systems with Intel CPUs. They have the same amount of RAM, same amount of cores, but the AMD CPUs are newer and have higher clock speeds. Is MEEP more optimized for Intel hardware?

**Comment by stevengj** [comment]
Clock speeds (which [haven't improved since 2005](http://cpudb.stanford.edu/visualize/clock_frequency)) haven't been a good proxy for performance for decades now…

For example, depending on what type of simulation you are doing, Meep's performance is affected strongly by the *speed* of your memory, not just on how much you have, which depends a lot on your system architecture.

**Reply by peterropac** [reply]
The simulations are large and I understand that memory speed will play a big role. The memory in the AMD machine is faster than the one in the Intel machine... I guess it is the architecture. 
I am surprised that the Intel machine can run the same simulation in almost half the time.

**Reply by stevengj** [reply]
The Intel CPUs also have wider SIMD instructions for floating-point operations, IIRC.

**Reply by stevengj** [reply, marked as answer]
It's also not just the RAM speed, but also the caches, and the memory coherency architecture on a multicore machine, … computers are complicated.

**Reply by peterropac** [reply]
They really are. I just thought since Intel CPUs were dominant for such a long time, some libraries were also mostly developed with Intel CPUs in mind, and that it might be the case with MEEP as well. Thank you for your replies.
